packages/query-engine/source/controllers/QueryEngineController.ts

```diff
@@ -0,0 +1,49 @@
+import mysql from "mysql2";
+
+import type {
+    MySQLConfiguration,
+    Resource,
+    Query,
+    QueryRequest,
+} from "@hypertool/common";
+
+import { QueryTemplateModel, ResourceModel } from "@hypertool/common";
+
+const queryEngineController = async (queryRequest: QueryRequest) => {
+    const query: Query = await QueryTemplateModel.findOne({
+        name: queryRequest.name,
+    }).exec();
+
+    const resource: Resource = await ResourceModel.findOne({
+        _id: query.resource,
+    }).exec();
+
+    const mySQLConfig: MySQLConfiguration = resource.mysql;
+
+    const connection = mysql.createConnection({
+        host: mySQLConfig.host,
+        port: mySQLConfig.port,
+        user: mySQLConfig.databaseUserName,
+        password: mySQLConfig.databasePassword,
+        database: mySQLConfig.databaseName,
+        namedPlaceholders:
+            typeof queryRequest.variables === "object" ? true : false,
+    });
+
+    let queryResult: any;
+    connection.execute(
+        query.content,
+        queryRequest.variables,
+        (error, results, fields) => {
+            if (error) {
+                queryResult = { error };
```

Comment:
1. Promisify the call to `connection.execute`.
2. Throw an exception when an error occurs instead of returning it. Create an appropriate error class. Add an error middleware that returns an error status based on the type of the exception.